### DISCUSSION TITLE
How do I link ImGui on macOS?

### DISCUSSION THREAD

**Discussion by mikesk8r** [ocornut/imgui#6411] [New Users Build/Link/Run issues ONLY!]
I keep getting an error and I have no idea what to do.
I have a OpenGL3 as my renderer and GLFW as my backend.

Error:
```
Undefined symbols for architecture x86_64:
  "ImGui::End()", referenced from:
      _main in main-023cd1.o
  "ImGui::Begin(char const*, bool*, int)", referenced from:
      _main in main-023cd1.o
ld: symbol(s) not found for architecture x86_64
```

**Comment by mikesk8r** [comment]
Is there some sort of library I need?
I have imgui and its dependencies in `/usr/local/include` and I have them included in my file.

**Comment by ocornut** [comment]
The symbol missing are ImGui::Begin() and ImGui::End() suggesting you are not LINKING with any object or library containing imgui code You need to compile imgui.cpp etc. and link the output into your app.

The simplest and recommended way is to include the .cpp files in your projects and have all imgui sources/includes in your project folder or as a git submodule. If somehow you have "installed" them in /usr/local/include then you need to link the corresponding library.

**Reply by mikesk8r** [reply]
Thanks
